Exceptions when on invalid (filtered) polygons

Hey,
Clarification: I don't know if this is entirely our fault or if Detour Crowd should be handling this kind of issue somehow.

1. When requestMoveTarget to a place which has a valid polygon, but is unwalkable for that agent because of Filter restrictions, the following NPE is thrown (We prevent this now, by including the filter into our polyRef calculation):

```
java.lang.NullPointerException
	at org.recast4j.detour.DetourCommon.vCopy(DetourCommon.java:101)
	at org.recast4j.detour.crowd.Crowd.checkPathValidity(Crowd.java:826)
	at org.recast4j.detour.crowd.Crowd.update(Crowd.java:875)
```
The `checkPathValidity` is why I'm reporting these issues, I suspect DetourCrowd wanted to handle that.

2. When adding an Agent to a Crowd but at a position which is unwalkable because of Filter restrictions:
```
Caused by: java.lang.NullPointerException
	at org.recast4j.detour.DetourCommon.vCopy(DetourCommon.java:101)
	at org.recast4j.detour.crowd.PathCorridor.reset(PathCorridor.java:201)
	at org.recast4j.detour.crowd.Crowd.addAgent(Crowd.java:389)
```

Thanks in Advance :)

--- a/detour/src/main/java/org/recast4j/detour/NavMeshQuery.java
+++ b/detour/src/main/java/org/recast4j/detour/NavMeshQuery.java
@@ -521,7 +521,7 @@ public class NavMeshQuery {
     /// @returns The status flags for the query.
     public Result<FindNearestPolyResult> findNearestPoly(float[] center, float[] halfExtents, QueryFilter filter) {
 
-        float[] nearestPt = null;
+        float[] nearestPt = new float[] {center[0], center[1], center[2]};
 
         // Get nearby polygons from proximity grid.
         Result<List<Long>> polysResult = queryPolygons(center, halfExtents, filter);

--- a/detour/src/test/java/org/recast4j/detour/FindNearestPolyTest.java
+++ b/detour/src/test/java/org/recast4j/detour/FindNearestPolyTest.java
@@ -44,4 +44,32 @@ public class FindNearestPolyTest extends AbstractDetourTest {
         }
 
     }
+
+    @Test
+    public void shouldReturnStartPosWhenNoPolyIsValid() {
+        QueryFilter filter = new QueryFilter() {
+            @Override
+            public boolean passFilter(long ref, MeshTile tile, Poly poly) {
+                return false;
+            }
+
+            @Override
+            public float getCost(float[] pa, float[] pb, long prevRef, MeshTile prevTile,
+                                 Poly prevPoly, long curRef, MeshTile curTile, Poly curPoly,
+                                 long nextRef, MeshTile nextTile, Poly nextPoly) {
+                return 0;
+            }
+        };
+        float[] extents = { 2, 4, 2 };
+        for (int i = 0; i < startRefs.length; i++) {
+            float[] startPos = startPoss[i];
+            Result<FindNearestPolyResult> poly = query.findNearestPoly(startPos, extents, filter);
+            assertTrue(poly.succeeded());
+            Assert.assertEquals(0L, poly.result.getNearestRef());
+            for (int v = 0; v < polyPos[i].length; v++) {
+                Assert.assertEquals(startPos[v], poly.result.getNearestPos()[v], 0.001f);
+            }
+        }
+
+    }
 }

